src/domain/noteSettings.service.ts

```diff
@@ -99,4 +107,36 @@ export default class NoteService {
   public async patchMemberRoleByUserId(id: NoteId, userId: UserId, newRole: MemberRole): Promise<MemberRole> {
     return await this.noteSettingsRepository.patchMemberRoleByUserId(id, userId, newRole);
   }
+
+  /**
+   * Set new parent for the note
+   *
+   * @param id - Note id
+   * @param parentURL - link to the new parent note
+   */
+  public async updateParent(id: NoteId, parentURL: string): Promise<void> {
+    // Regex matches any characters between '/note/' and the next slash or end of string
+    const regex = /\/note\/([^/]+)/;
+
+    const matches = parentURL.match(regex);
+
+    if (matches === null) {
+      throw new Error('Invalid parent URL');
+    }
+
+    // Extracts the ID from the URL. The ID is matches[1] as matches[0] is the full match
+    const parentId = matches[1];
+
+    const noteIdPattern = /^[a-zA-Z0-9-_]{10}$/;
+
+    if (!noteIdPattern.test(parentId)) {
+      throw new Error('Invalid parent ID');
+    }
+
+    const isUpdated = await this.noteRepository.updateParent(id, parentId);
```

Comment:
why `updateParent` not in the noteSettingsRepository?

Reply:
Because it is not part of NoteSettings. In api it also located in noteRepository